core/util.py

```diff
@@ -89,3 +95,37 @@ def get_project_number_by_id(project_id: str, active_project_ids: list[str]) ->
     project_number = active_project_ids.index(project_id) + 1
 
     return project_number
+
+
+def load_example_data():
+    """
+    Load example data into DB.
+
+    Intended to be used only for local example / test data. Will be loaded into current app context.
+    DO NOT use this util in any of the main app code as this will load FAKE example data into the DB.
+
+    NOTE data loaded this way is NOT validated against any of the schema rules, and is intended for testing DB
+    behaviour only (not data context / quality).
+    """
+    # load in table data from csv. File names match table definitions for convenience.
+    for table in [
+        "submission_dim",
+        "organisation_dim",
+        "programme_dim",
+        "project_dim",
+        "output_dim",
+        "outcome_dim",
+        "programme_progress",
+        "place_detail",
+        "funding_question",
+        "project_progress",
+        "funding",
+        "funding_comment",
+        "private_investment",
+        "output_data",
+        "outcome_data",
+        "risk_register",
+    ]:
+        table_df = pd.read_csv(resources / f"{table}.csv")
+        table_df.to_sql(table, con=db.session.connection(), index=False, index_label="id", if_exists="append")
```

Comment:
Just clarifying my understanding. With ```if_exists="append"``` we expect for the purposes of this function that there may already be data in the db, and that whatever data is currently in the db we don't want to remove or replace. Would  ```if_exists="replace"``` not be better if we want to ensure a clean dataset for the purposes of testing?

Reply:
I think I'd prefer to keep it as is, as the whole point of this refactor is there should be either a clean database or some pre-existing populated data (depending on the fixture used). If somehow test data "leaks" through from another test via incorrect implementation of fixture scope, using "append" rather than "replace" is more likely to cause this to error, at least in my mind (it falling over is preferable to just hiding the problem, so at least we spot it happening). This hopefully would help avoid the kind of situation we had before where tests were accidentally sharing DB changes from each other?

Reply:
"help avoid the kind of situation we had before where tests were accidentally sharing DB changes from each other"

Wouldn't using replace be more likely to ensure this rather than append, as append will retain db changes from one test to another?

Reply:
Oh, sorry, just re-read. We want the error the occur rather than masking it. Makes sense.

Reply:
Also, replace drops the entire table. Which means that our module scope seed fixture will have effectively been wiped during a single test and the other tests that need that seed data will not be able to see it. It also means we then recreate the table each test, which is effectively what the refactor was replacing (with only creating the table at module setup)